shell/common/api/electron_api_native_image_mac.mm

```diff
@@ -34,6 +39,64 @@
   return def;
 }
 
+// static
+v8::Local<v8::Promise> NativeImage::CreateThumbnailFromPath(
+    v8::Isolate* isolate,
+    const base::FilePath& path,
+    const gfx::Size& size) {
+  gin_helper::Promise<gfx::Image> promise(isolate);
+  v8::Local<v8::Promise> handle = promise.GetHandle();
+
+  if (size.width() <= 0) {
+    promise.RejectWithErrorMessage(
+        "invalid width, please enter a positive number");
+    return handle;
+  }
+
+  if (size.height() <= 0) {
+    promise.RejectWithErrorMessage(
+        "invalid height, please enter a positive number");
+    return handle;
+  }
+
+  if (!path.IsAbsolute()) {
+    promise.RejectWithErrorMessage("path must be absolute");
+    return handle;
+  }
```

Comment:
We don't need 2-3 copies of the input sanitizing code. Better to have one function in electron-api-native-image.cc that does these checks, then delegates the work out to per-platform functions that take presanitized input.

Reply:
This is removed in the latest commit, it was my mistake where I was unaware that gfx::Size does not allow negative inputs